tools/c7n_gcp/c7n_gcp/resources/cloudtasks.py

```diff
@@ -0,0 +1,378 @@
+from c7n_gcp.actions import MethodAction
+from c7n_gcp.provider import resources
+from c7n_gcp.query import QueryResourceManager, TypeInfo, ChildResourceManager, ChildTypeInfo
+
+from c7n.utils import local_session, type_schema
+
+
+@resources.register('cloudtasks-location')
+class CloudTasksLocation(QueryResourceManager):
+    """GCP resource: https://cloud.google.com/tasks/docs/reference/rest/v2/projects.locations
+    This one is present only as the parent of other resources.
+    """
+    def get_resource_query(self):
+        return {'name': 'projects/%s' % local_session(self.session_factory).get_default_project()}
+
+    class resource_type(TypeInfo):
+        service = 'cloudtasks'
+        version = 'v2'
+        component = 'projects.locations'
+        enum_spec = ('list', 'locations[]', None)
+        scope = None
+        id = 'name'
+
+        @staticmethod
+        def get(client, resource_info):
+            return client.execute_query('get', {'name': resource_info['name']})
+
+
+@resources.register('cloudtasks-queue')
+class CloudTasksQueue(ChildResourceManager):
+    """GCP resource:
+    https://cloud.google.com/tasks/docs/reference/rest/v2/projects.locations.queues
+    """
+    def _get_child_enum_args(self, parent_instance):
+        return {'parent': parent_instance['name']}
+
+    def _get_parent_resource_info(self, child_instance):
+        return {'name': re.match('(projects/.*?/locations/.*?)/queues/.*',
+                                 child_instance['name']).group(1)}
+
+    class resource_type(ChildTypeInfo):
+        service = 'cloudtasks'
+        version = 'v2'
+        component = 'projects.locations.queues'
+        enum_spec = ('list', 'queues[]', None)
+        scope = None
+        get_requires_event = True
+        id = 'name'
+        parent_spec = {
+            'resource': 'cloudtasks-location'
+        }
+
+        @staticmethod
+        def get(client, event):
+            return client.execute_query(
+                'get', {'name': jmespath.search('protoPayload.response.name', event)})
+
+
+@CloudTasksQueue.action_registry.register('delete')
+class CloudTaskQueueDelete(MethodAction):
+    """
+    `Deletes <https://cloud.google.com/tasks/docs/reference/rest/v2/projects.locations.queues
+    /delete>`_ a Task Queue. The action does not specify any parameters.
+
+    Example:
+    .. code-block:: yaml
+
+        policies:
+          - name: gcp-cloudtasks-queue-set
+            resource: gcp.cloudtasks-queue
+            filters:
+              - type: value
+                key: name
+                value: projects/target-project/locations/target-location/queues/queue-to-delete
+            actions:
+              - type: delete
+    """
+    schema = type_schema('delete')
+    method_spec = {'op': 'delete'}
+
+    def get_resource_params(self, m, r):
+        return {'name': r['name']}
+
+
+@CloudTasksQueue.action_registry.register('set')
+class CloudTaskQueueSet(MethodAction):
+    """
```

Comment:
i thought we had agreed to a prioritized list of things to work on, this really doesn't have governance compliance concerns and is really far down a list. 

ie. can we focus dev efforts on things actually useful for users. this resource probably shouldn't exist, everything here is an applicatin provisioning concern with no real world governance issues.

Reply:
We can exclude cloud tasks from the implemented resources for now.

During the meeting, this resource was under review because I believe you thought it was analogous to SQS messaging in AWS, and the decision on whether to include was pending investigation into this assertion. Upon investigation from my side, the resource was not analogous to SQS resources, and so we included it.

I do agree that the use case should have been included in the PR description.